### PR TITLE
Restored create SD page, SD details page and added new sector script

### PR DESCRIPTION
**Pull request by jim68000**
~Does not save yet - looks like Django is expecting an event even though they are optional~

Now saves and displays

### CODE REVIEW

**Comment by reupen** [comment]
Getting this error when trying to create a service delivery for a contact:

```
TypeError: Cannot read property 'company' of undefined
    at data-hub-fe-beta2/src/controllers/servicedeliverycontroller.js:53:55
```

**Comment by jim68000** [comment]
s/contactId/contact/g did the trick for that one @reupen

File: src/controllers/servicedeliverycontroller.js [hunk 1]
```diff
@@ -18,7 +18,13 @@ function getCommon (req, res, next) {
   Q.spawn(function * () {
     try {
       const token = req.session.token
-      res.locals.serviceDelivery = yield serviceDeliveryService.getHydratedServiceDelivery(token, req.params.serviceDeliveryId)
+      // if we are creating a new service delivery then the id comes through as edit
+      // @TODO make the routes a bit more sensible
```
Comment (ztolley) [inline src/controllers/servicedeliverycontroller.js]: Is there a card for this?

File: src/controllers/servicedeliverycontroller.js [hunk 2]
```diff
@@ -77,14 +81,15 @@ function postServiceDeliveryEdit (req, res, next) {
       delete req.body.date_month
       delete req.body.date_day
 
-      req.body = nullEmptyFields(req.body)
+      // v2 endpoint rejects nulls
+      req.body = deleteNulls(nullEmptyFields(req.body))
```
Comment (ztolley) [inline src/controllers/servicedeliverycontroller.js]: What happens if we want to clear the value in a field? Say if you want to to so there is no longer any notes (though I know technically every field is needed)

File: src/lib/propertyhelpers.js [hunk 3]
```diff
@@ -18,6 +18,18 @@ function nullEmptyFields (data) {
   return cleanedObject
 }
 
+function deleteNulls (data) {
```
Comment (ztolley) [inline src/lib/propertyhelpers.js]: See previous comment. Deleting null fields may in some situations result in an edit not being committed as the user intended.

File: test/lib/controllerutils.test.js [hunk 4]
```diff
@@ -24,3 +24,100 @@ describe('isBlank', function () {
...
+    const source = [
+      {
+        'detail': 'This combination of service and service provider does not exist.',
+        'source': {
+          'pointer': '/data/relationships/service'
+        }
+      }
+    ]
+    const actual = transformV2Errors(source)
+    expect(actual.Alert).to.exist
```
Comment (ztolley) [inline test/lib/controllerutils.test.js]: Why is alert Capital?

File: test/lib/controllerutils.test.js [hunk 5]
```diff
@@ -24,3 +24,100 @@ describe('isBlank', function () {
...
+      {
+        'detail': "{'data': {'type': 'Country'}} has no key id",
+        'source': {
+          'pointer': '/data/relationships/country_of_interest'
+        }
+      }
+    ]
+    const actual = transformV2Errors(source)
+    expect((Object.keys(actual)).length).to.equal(9)
+    expect(actual.subject).to.be.defined
```
Comment (ztolley) [inline test/lib/controllerutils.test.js]: I use 

expect(actual).to.have.property('subject') 

but I'm guessing thats just my style.

File: src/controllers/servicedeliverycontroller.js (hunk 1 above)
Comment (jim68000) [inline src/controllers/servicedeliverycontroller.js]: No, I'll add one to backlog as debt

File: src/controllers/servicedeliverycontroller.js (hunk 2 above)
Comment (jim68000) [inline src/controllers/servicedeliverycontroller.js]: That's not the way the API behaves. Subject, notes and date are mandatory fields, so the server rejects nulls for those, and if you try and send a null in the relationship fields it errors out.

**Comment by jim68000** [comment]
Yeah, I realised what I actually need is integration tests so I can check that all the entrance points work with the pernicketty server

**Pull request merged**